### DISCUSSION TITLE
Chat is not working and getting unable to get local issuer certificate

### DISCUSSION THREAD

**Discussion by venkateshchidambaranathan** [Genuitec/Copilot4Eclipse-community#44] [Help] [answered]
I am getting **unable to get local issuer certificate** json response in the back end. If i start interact with the chat it keep trying and no responses showed.

Language Server Log Json Responses:
`[2024-11-01T15:12:41.597652500-04:00] LANGUAGE_SERVER_TO_LSP4E com.genuitec.copilot4eclipse.copilot:
{"jsonrpc":"2.0","method":"statusNotification","params":{"status":"Warning","message":"unable to get local issuer certificate"}}`

"steps":[{"id":"generate-response","title":"Generating response","status":"failed","error":{"message":"Sorry, an error occurred while generating a response."}}]}}}

Eclipse : 2024-03 / 2024-06
Plug in Version : 1.3.1

Appreciate your help to resolve this issue.

**Comment by wayneparrott** [comment]
Are you operating Copilot4Eclipse from behind a corporate proxy or firewall?
If yes, I'm thinking there is a self-signed cert configured on the proxy for HTTPS Inspection. 

W

**Reply by venkateshchidambaranathan** [reply]
@wayneparrott thank you. yes its behind corporate proxy or firewall.
Also I have IntelliJ and Visual Studio in that git hub copilot is working fine.
Is there any special settings required for Eclipse alone ?

Note: I have **node_extra_ca_certs** environment variable that poiting to the corresponding file.

**Comment by venkateshchidambaranathan** [comment]
@wayneparrott is there any way we can try with older version Coplio4Eclipse?. We think this might be latest plugin version issue. 
One of the team member they installed the version **Version: 1.2.0.EA1** earlier and its perfectly working fine as of now. 

Note: We are using the Github copilot business subscription.

**Reply by wayneparrott** [reply]
Contact me directly at support@genuitec.com. We can discuss. Please address note to my attention (Wayne) and include a link to this thread.

W

**Reply by venkateshchidambaranathan** [reply]
@wayneparrott thanks. I have sent an email.

**Comment by venkateshchidambaranathan** [comment]
@wayneparrott i seen this forum https://www.genuitec.com/forums/topic/cant-authorize-github-copilot/
and we are getting the same issues. The suggested answer in the post was not working. 

This was the link provided in the forum.
https://www.genuitec.com/products/copilot4eclipse/docs/resources/technotes/firewall-settings.

Could you please suggest

**Comment by venkateshchidambaranathan** [comment, marked as answer]
@wayneparrott we have done the below steps after that its started working fine.

1. Select the company certificate for Copilot and  Install Certificate… . The Certificate Import wizard appears.
2. In the wizard, choose Next . Then, when you are prompted for the Certificate Store, choose Place all certificates in the following store
 Select the **Trusted Root Certification Authorities** store
3. Complete the remaining steps of the wizard and click Finish

**Reply by wayneparrott** [reply]
Thanks for the update. I'm glad this work for you as we manage similar internal certs/CA using the windows certmanager and mac keystore for our proxy. Once setup it is painless thereafter.

W
Copilot4Eclipse Team

**Comment by venkateshchidambaranathan** [comment]
@wayneparrott during the triage we found below observations. 
Eclipse Copilot is using the IntelliJ login auth token.

Steps:
1. Intellj IDE -  Login Copilot
2. Eclipse install the plugin (Automatically logged in without any sign in from eclipse)

Intellij is creating the **hosts.json** in the folder C:\Users\USER\AppData\Local\**github-copilot** after the login. Eclipse Copilot is referring this file.

When we login Copilot via Eclipse its creating **apps.json** in the same folder. This is the login file for eclipse.

**Reply by wayneparrott** [reply]
The api Copilot4Eclipse uses is common to Copilot extensions for IJ, vim and others.
